### ISSUE TITLE
ClassNotFoundException: JavaDialectConfiguration

### ISSUE THREAD

**Issue by tdiesler** [wildfly-extras/wildfly-camel#2831]
Camel3: [drools-7.23.0.Final](https://github.com/apache/camel/blob/master/parent/pom.xml#L184)

CrossRef: https://issues.apache.org/jira/browse/CAMEL-13654

**Comment by tdiesler** [comment]
drools-7.23 uses this code ...

```
    public void setDroolsClassLoader(ClassLoader droolsClassLoader) {
        if (getParent() != droolsClassLoader && isOsgiClassLoader(droolsClassLoader)) {
            this.droolsClassLoader = droolsClassLoader;
            if (CACHE_NON_EXISTING_CLASSES) {
                nonExistingClasses.clear();
            }
        }
    }
```

**Comment by tdiesler** [comment]
Reported [here](https://github.com/kiegroup/drools/commit/85c5308031a84a0ac4086c1df363bb1b4b3c6af9#r33962268)

**Comment by mariofusco** [comment]
I reported the problem [here](https://issues.jboss.org/browse/DROOLS-4416) and fixed it with [this commit](https://github.com/kiegroup/drools/commit/06a31bd9349487e64edb384875430df369489c87). Fix will be available with drools 7.26.0.Final
